cmd/tempo-cli/cmd-rewrite-blocks.go

```diff
@@ -38,51 +39,72 @@ func (cmd *dropTraceCmd) Run(ctx *globalOptions) error {
 		fmt.Println("")
 	}
 
-	r, w, c, err := loadBackend(&cmd.backendOptions, ctx)
-	if err != nil {
-		return err
-	}
+	ctx := context.Background()
 
-	id, err := util.HexStringToTraceID(cmd.TraceID)
+	r, w, c, err := loadBackend(&cmd.backendOptions, opts)
 	if err != nil {
 		return err
 	}
 
-	blocks, err := blocksWithTraceID(context.Background(), r, cmd.TenantID, id)
-	if err != nil {
-		return err
+	type pair struct {
+		traceIDs  []common.ID
+		blockMeta *backend.BlockMeta
 	}
+	tracesByBlock := map[backend.UUID]pair{}
 
-	if len(blocks) == 0 {
-		fmt.Println("\ntrace not found in any block. aborting")
-		return nil
-	}
+	// Group trace IDs by blocks
+	ids := strings.Split(cmd.TraceID, ",")
+	for _, id := range ids {
+		traceID, err := util.HexStringToTraceID(id)
+		if err != nil {
+			return err
+		}
 
-	// print out blocks that have the trace id
-	fmt.Println("\n\ntrace found in:")
-	for _, block := range blocks {
-		fmt.Printf("  %v sz: %d traces: %d\n", block.BlockID, block.Size_, block.TotalObjects)
-	}
+		// It might be significantly improved if common.BackendBlock supported bulk searches.
+		blocks, err := blocksWithTraceID(ctx, r, cmd.TenantID, traceID)
+		if err != nil {
+			return err
+		}
 
-	if !cmd.DropTrace {
-		fmt.Println("**not dropping trace, use --drop-trace to actually drop**")
-		return nil
+		if len(blocks) == 0 {
+			fmt.Printf("\ntrace %s not found in any block. skipping", traceID)
+		}
+		for _, block := range blocks {
+			p, ok := tracesByBlock[block.BlockID]
+			if !ok {
+				p = pair{blockMeta: block}
+			}
+			p.traceIDs = append(p.traceIDs, traceID)
+			tracesByBlock[block.BlockID] = p
+		}
 	}
 
-	fmt.Println("rewriting blocks:")
-	for _, block := range blocks {
-		fmt.Printf("  rewriting %v\n", block.BlockID)
-		newBlock, err := rewriteBlock(context.Background(), r, w, block, id)
+	// Remove traces from blocks
+	for _, p := range tracesByBlock {
+		// print out trace IDs to be removed in the block
+		{
+			traceIDs := make([]string, len(p.traceIDs))
+			for i, tid := range p.traceIDs {
+				traceIDs[i] = util.SpanIDToHexString(tid)
```

Comment:
```suggestion
				traceIDs[i] = util.TraceIDToHexString(tid)
```

we are dealing with TraceIDs and not SpanIDs, so we want to use `TraceIDToHexString` here, we were using `HexStringToTraceID(cmd.TraceID)` here.

docs:
> - SpanIDToHexString converts a span ID to its string representation and WITHOUT removing any leading zeros.
> - TraceIDToHexString converts a trace ID to its string representation and removes any leading zeros.